docs/platform/connector-development/connector-builder-ui/authentication.md

```diff
@@ -93,57 +97,128 @@ curl -X GET \
 
-The configuration consists of three main components:
+- "JWT authentication"
+- "JSON Web Token"
+- "Bearer JWT"
+- "Signed tokens"
+
+JWT authentication generates a signed token using a secret key and algorithm, which is then sent with each request. The connector builder supports various JWT algorithms and configuration options.
+
+#### Configuration
+
+The JWT authentication method requires the following configuration:
+
+- **Secret Key** - The secret key used to sign the JWT token. This can be set via "Testing values" in the connector builder and by the end user when configuring this connector as a Source.
+- **Algorithm** - The algorithm used to sign the token. Supported algorithms include:
+  - HS256 (HMAC with SHA-256)
+  - HS384 (HMAC with SHA-384) 
+  - HS512 (HMAC with SHA-512)
+  - RS256 (RSA with SHA-256)
+  - RS384 (RSA with SHA-384)
+  - RS512 (RSA with SHA-512)
+  - ES256 (ECDSA with SHA-256)
+  - ES384 (ECDSA with SHA-384)
+  - ES512 (ECDSA with SHA-512)
+- **Base64 Encode Secret Key** - When enabled, the secret key will be base64 encoded before being used to sign the JWT. Only enable this if required by the API.
+- **Header Prefix** - The prefix to use in the Authorization header (defaults to "Bearer")
+- **JWT Headers** - Additional headers to include in the JWT header object
+- **JWT Payload** - Additional properties to include in the JWT payload
+- **Additional JWT Headers** - Extra headers to include with the JWT headers object
+- **Additional JWT Payload** - Extra properties to add to the JWT payload
+
+#### Example
+
+When fetching records, the JWT token is generated and sent as part of the `Authorization` header:
+
+```
+curl -X GET \
+  -H "Authorization: Bearer <jwt-token>" \
+  https://api.example.com/<stream path>
+```
+
+The JWT token is automatically generated by the connector using the configured secret key and algorithm, and includes any additional headers or payload properties you've specified.
+
+### OAuth 2.0 (Declarative)
+
+Declarative OAuth 2.0 provides a comprehensive way to configure any OAuth 2.0 flow by describing the exact structure and behavior of the authentication endpoints. This is the recommended approach for implementing OAuth 2.0 authentication in the Connector Builder, as it supports the complete OAuth flow including user consent and automatic token management.
+
+#### Key Features
+
+- **Complete OAuth 2.0 Flow** - Supports the full authorization code flow with user consent
+- **Flexible Configuration** - Accommodates OAuth 2.0 implementations that deviate from standard specifications
+- **Automatic Token Management** - Handles access token refresh automatically
+- **Custom Parameter Support** - Allows injection of custom parameters and headers at each step
+
+#### Configuration Components
+
+The Declarative OAuth 2.0 configuration consists of three main components:
```

Comment:
This whole section will probably need to be rewritten to reflect what I mentioned above